kotlinx-coroutines-core/jvm/src/scheduling/CoroutineScheduler.kt

```diff
@@ -456,12 +461,14 @@ internal class CoroutineScheduler(
         }
     }
 
-    /*
+    /**
      * Returns the number of CPU workers after this function (including new worker) or
      * 0 if no worker was created.
      */
     private fun createNewWorker(): Int {
-        synchronized(workers) {
+        var worker: Worker
+        // kotlin.s to have contracts working
+        val result = kotlin.synchronized(workers) {
```

Comment:
This one is totally bonkers.

There is no way to mark `expect` declaration with contracts, but if I mark actual `kx.c.synchronized`  with contacts, compiler start reporting "default value is unused" when compiling common code, triggering failure due to `-Werror`

Reply:
"Default value is unused" seems to be a correct warning that we could just fix. I see two ways to do that:
* Make `synchronized` have the contract on all platforms (undesirable, as the IDE still won't reflect the warnings corresponding to the contracts),
* Add a wrapper function with the contract (example: https://gist.github.com/dkhalanskyjb/43a6668954cd7ee781473339c5a03c32). Can be done like in my patch or by adding a `synchronizedWithContract` and using that where applicable and keeping `synchronized` as is.